### ISSUE TITLE
insert binary package - bug when detecting pkgname

### ISSUE THREAD

**Issue by jangorecki** [eddelbuettel/drat#39]
Seems to be a bug in [getPackageInfo](https://github.com/eddelbuettel/drat/blob/80873a3164c720fed71938e02194b9380357650b/R/insertPackage.R#L187)
In regex in linked line seems that `getPackageInfo` is not handling dots in package name.
```r
gsub("^([a-zA-Z0-9]*)_.*", "\\1", "my.pkg_1.0.0.zip") # should return "my.pkg"
# [1] "my.pkg_1.0.0.zip"
```
Which later results in error.
To reproduce take the `zip` binaries of package with dots in the name and try `insertPackage()` on it.

```
R version 3.2.2 (2015-08-14)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.3 LTS

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8     LC_MONETARY=en_GB.UTF-8   
 [6] LC_MESSAGES=en_GB.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] drat_0.1.0

loaded via a namespace (and not attached):
[1] tools_3.2.2  git2r_0.10.1
```

**Comment by eddelbuettel** [comment]
Can you cook up a patched reg.exp?

**Comment by jangorecki** [comment]
I'm not so good in regex, maybe `strsplit`?
```r
f = function(file){
    pkgname = strsplit(file, "_", fixed=TRUE)[[1L]]
    paste(pkgname[-length(pkgname)], collapse="_")    
}
sapply(c("my_pkg_1.0.0.zip","my.pkg_1.0.0.zip"), f)
# my_pkg_1.0.0.zip my.pkg_1.0.0.zip 
#         "my_pkg"         "my.pkg" 
```

**Comment by eddelbuettel** [comment]
Try adding `._` in the first square bracket term here: `gsub("^([a-zA-Z0-9]*)_.*", "\\1", "my_pkg_1.0.0.zip")`

It may break detection of the package version though.

**Comment by jangorecki** [comment]
fix already merged in #40
